bblfsh/compat.py

```diff
@@ -0,0 +1,228 @@
+    def __init__(self, ctx: ResultContext, filename: str = "") -> None:
+        self._res_context = ctx
+        self._filename = filename
+
+    @property
+    def uast(self) -> Node:
+        return self._res_context.uast
+
+    @property
+    def ast(self) -> Node:
+        return self._res_context.ast
+
+    @property
+    def ctx(self) -> ResultContext:
+        return self._res_context
+
+    @property
+    def elapsed(self) -> int:
+        # FIXME(juanjux): check if the can get this or measure ourselves
+        return -1
+
+    @property
+    def language(self) -> str:
+        return self._res_context.language
+
+    @property
+    def filename(self) -> str:
+        return self._filename
+
+    @property
+    def DESCRIPTOR(self) -> Any:
+        return self._res_context.ctx.DESCRIPTOR
+
+    @property
+    def errors(selfs) -> List:
+        # ParseResponse would have raised an exception on errors
+        return []
+
+
+class CompatBblfshClient:
+    def __init__(self, endpoint: Union[str, grpc.Channel]) -> None:
+        self._bblfsh_cli = bcli.BblfshClient(endpoint)
+
+        self._channel = self._bblfsh_cli._channel
+        self._stub_v1 = self._bblfsh_cli._stub_v1
+        self._stub_v2 = self._bblfsh_cli._stub_v2
+
+    def _parse(self, filename: str, language: str = None, contents: str = None,
+               timeout: float = None,
+               mode: ModeType = Mode.Value('ANNOTATED')) -> CompatParseResponse:
+
+        if timeout is not None:
+            timeout = int(timeout)
+
+        res = self._bblfsh_cli.parse(filename, language, contents,
+                                     mode=mode, timeout=timeout)
+        return CompatParseResponse(res, filename)
+
+    def parse(self, filename: str, language: str = None, contents: str = None,
+              timeout: float = None) -> CompatParseResponse:
+
+        return self._parse(filename, language, contents, timeout,
+                           Mode.Value('ANNOTATED'))
+
+    def native_parse(self, filename: str, language: str = None,
+                     contents: str = None,
+                     timeout: float = None) -> CompatParseResponse:
+
+        return self._parse(filename, language, contents, timeout,
+                           Mode.Value('NATIVE'))
+
+    def supported_languages(self) -> List[str]:
+        return self._bblfsh_cli.supported_languages()
+
+    def version(self) -> VersionResponse:
+        return self._bblfsh_cli.version()
+
+    def close(self) -> None:
+        return self._bblfsh_cli.close()
+
+
+class CompatNodeIterator:
+    def __init__(
+            self,
+            nodeit: NodeIterator,
+            only_nodes: bool = False
+    ) -> None:
+        self._nodeit = nodeit
+        self._only_nodes = only_nodes
+        # Used to forward calls of the old Node object
+        # Check if this, and properties(), are needed
+        self._last_node: Optional[Node] = None
+
+    def __iter__(self) -> 'CompatNodeIterator':
```

Comment:
Why a string instead of the class itself? User-defined classes _should_ work as type annotations, but maybe there is something else going on in this case?

Reply:
You use a string with mypy annotations when it's a reference to the enclosing class (as is the case here) or other class that is not defined yet.

Reply:
Makes sense, thanks for the explanation!